src/coreclr/vm/methodtable.cpp

```diff
@@ -6732,13 +6720,13 @@ MethodDesc *MethodTable::MethodDataObject::GetImplMethodDesc(UINT32 slotNumber)
     if (pMDRet == NULL)
     {
         _ASSERTE(slotNumber < GetNumVirtuals());
-        pMDRet = m_pDeclMT->GetMethodDescForSlot(slotNumber);
+        pMDRet = m_pDeclMT->GetMethodDescForSlot_NoThrow(slotNumber);
```

Comment:
The contract was updated to `THROWS`, but we're calling the "NoThrow" variant. That seems odd.

Reply:
I'm still unclear on this code path. The only changes here are going from `NOTHROW` to `THROWS` and adding a method that specifically has the `NoThrow` in the name. That is unexpected. What is the "NoThrow" doing that requires the contract change?

Reply:
Ah, this is a case where I believe I changed to a NoThrow variant after changing to a THROWS contract. I'll swap this back and verify this is true.